### PR TITLE
fix owncloud user enumeration vulnerbility #23595

### PR DESCRIPTION
**Pull request by otih**
still more code cleanup needed

### CODE REVIEW

**Comment by PVince81** [comment]
@Peter-Prochaska

**Comment by Peter-Prochaska** [comment]
Yes?

File: core/Controller/LostController.php [hunk 1]
```diff
@@ -236,19 +236,10 @@ public function setPassword($token, $userId, $password, $proceed) {
 	 * @throws \Exception
 	 */
 	protected function sendEmail($user) {
-		if (!$this->userManager->userExists($user)) {
```
Comment (PVince81) [inline core/Controller/LostController.php]: keep the check but `return;` directly

File: core/Controller/LostController.php [hunk 2]
```diff
@@ -236,19 +236,10 @@ public function setPassword($token, $userId, $password, $proceed) {
...
-			throw new \Exception($this->l10n->t('Couldn\'t send reset email. Please make sure your username is correct.'));
-		}
 
 		$userObject = $this->userManager->get($user);
 		$email = $userObject->getEMailAddress();
 
-		if (empty($email)) {
-			throw new \Exception(
```
Comment (PVince81) [inline core/Controller/LostController.php]: same here

File: core/Controller/LostController.php (hunk 1 above)
Comment (PVince81) [inline core/Controller/LostController.php]: actually, don't return but render the template without sending out the email.

Comment (Peter-Prochaska) [inline core/Controller/LostController.php]: why not:

if ($this->userManager->userExists($user)) {
                $userObject = $this->userManager->get($user);
  		$email = $userObject->getEMailAddress();

File: core/Controller/LostController.php (hunk 2 above)
Comment (Peter-Prochaska) [inline core/Controller/LostController.php]: and here:

if (!empty($email)) {
...send email...

and after all, show a message "if that user exists then an email was sent", no matter if the user exists or not?

File: core/Controller/LostController.php (hunk 1 above)
Comment (VicDeo) [inline core/Controller/LostController.php]: I'd say these checks should create a log entry with severity `info` that contains a remote user IP in order to setup a fail2ban filter.

**Comment by kdslkdsaldsal** [comment]
Fixes: https://github.com/owncloud/core/issues/23595

**Comment by PVince81** [comment]
@otih any update ? Will you be able to finish this (+ backports) before the RC1 next week ?

If not, let us know how we can help.

**Comment by PVince81** [comment]
@Peter-Prochaska can you take over and help finish this ?

**Comment by jvillafanez** [comment]
I'd rather log something here so the admin can check what's happening.

**Comment by PVince81** [comment]
Superseded by https://github.com/owncloud/core/pull/27011

**Pull request closed without merging**